ol_sale_cost_workup/data/mail_template.xml

```diff
@@ -0,0 +1,20 @@
+<odoo>
+    <function name="write" model="ir.model.data">
```

Comment:
@Vandan-OSI why are we modifying this email template? The cost workup report doesn't have any email requirements in it so I don't think we need to touch any mail templates.

Reply:
> @Vandan-OSI why are we modifying this email template? The cost workup report doesn't have any email requirements in it so I don't think we need to touch any mail templates.

Because it is mentioned in the design.![image](https://github.com/user-attachments/assets/c3c086fd-5684-485f-9d30-c1c9c6f8041f)

Reply:
@Vandan-OSI I went ahead and updated the code to address this. The report is internal only so we don't want it sent to customer, so it gets generated when the user sends the quote but isn't part of the email itself. It should just get attached to the sale order as an attachment so the user could view it if they wanted to. I added this functionality and it will also generate the report when the SO gets locked. I added a server action to allow a user to generate it on the SO too.

We are missing tests, can you please work on those? I may merge this tomorrow so users can test but continue with the tests and create a new PR if I do merge this before you finish them. Thanks.